src/Data/Vec/Properties.agda

```diff
@@ -1100,6 +1162,31 @@ toList∘fromList : (xs : List A) → toList (fromList xs) ≡ xs
 toList∘fromList List.[]       = refl
 toList∘fromList (x List.∷ xs) = cong (x List.∷_) (toList∘fromList xs)
 
+toList-cast : ∀ .(eq : m ≡ n) (xs : Vec A m) → toList (cast eq xs) ≡ toList xs
+toList-cast {n = zero}  eq []       = refl
+toList-cast {n = suc _} eq (x ∷ xs) =
+  cong (x List.∷_) (toList-cast (cong pred eq) xs)
+
+cast-fromList : ∀ {xs ys : List A} (eq : xs ≡ ys) →
```

Comment:
I don't think this lemma is needed: one way to view it is as the composition of `cong fromList` and `cast-is-id`, or am I missing something?

Reply:
Yes, it is sort of `cong`ing `fromList` onto the equality, but the dependency in the types makes it to be not quite just adding `fromList` after `cast-is-id`. That is, the `refl` in your implementation substituted two types and rather than just one. My best attempt at rewriting it using the standard library is:

```agda
cast-fromList′ : ∀ {a} {A : Set a} {xs ys : List A} →
                 (eq : xs ≡ ys) → cast (cong List.length eq) (fromList xs) ≡ fromList ys
cast-fromList′ {xs = xs} eq =
  dsubst₂ (λ ys eq → cast eq (fromList xs) ≡ fromList ys) {y₁ = refl} eq refl (cast-is-id _ (fromList xs))
```
which I think is rather difficult to write in an inline style (for usages in equational reasoning). Therefore I think it is worth having a separate lemma.

There's a minor issue on how the lemma should compute. The implementation in the PR computes on the list, whereas the `cong`-like implementation computes on `eq`. I don't have strong opinions on which should be the case.